addons/test_mail_full/tests/test_mail_performance.py

```diff
@@ -95,3 +104,192 @@ def test_message_post_w_followers(self):
+                        'res_id': record.id,
+                        'res_model': record._name,
+                    })
+                    for att_idx in range(2)
+                ],
+                'author_id': record.customer_id.id,
+                'body': f'<p>Test {msg_idx}</p>',
+                'date': datetime(2023, 5, 15, 10, 30, 5),
+                'email_from': record.customer_id.email_formatted,
+                'link_preview_ids': [
+                    (0, 0, {
+                        'source_url': 'https://www.odoo.com',
+                    }), (0, 0, {
+                        'source_url': 'https://www.example.com',
+                    }),
+                ],
+                'notification_ids': [
+                    (0, 0, {
+                        'is_read': False,
+                        'notification_type': 'inbox',
+                        'res_partner_id': cls.customers[(msg_idx * 2)].id,
+                    }),
+                    (0, 0, {
+                        'is_read': True,
+                        'notification_type': 'email',
+                        'notification_status': 'sent',
+                        'res_partner_id': cls.customers[(msg_idx * 2) + 1].id,
+                    }),
+                ],
+                'message_type': 'comment',
+                'model': record._name,
+                'partner_ids': [
+                    (4, cls.customers[(msg_idx * 2)].id),
+                    (4, cls.customers[record_idx].id),
+                ],
+                'reaction_ids': [
+                    (0, 0, {
+                        'content': 'https://www.odoo.com',
+                        'partner_id': cls.customers[(msg_idx * 2) + 1].id
+                    }), (0, 0, {
+                        'content': 'https://www.example.com',
+                        'partner_id': cls.customers[record_idx].id
+                    }),
+                ],
+                'res_id': record.id,
+                'subject': f'Test Rating {msg_idx}',
+                'subtype_id': comment_subtype_id,
+                'starred_partner_ids': [
+                    (4, cls.customers[(msg_idx * 2)].id),
+                    (4, cls.customers[(msg_idx * 2) + 1].id),
+                ],
+                'tracking_value_ids': [
+                    (0, 0, {
+                        'field': user_id_field.id,
+                        'field_desc': 'Responsible',
+                        'new_value_char': 'new 1',
+                        'new_value_integer': record.user_id.id,
+                        'old_value_char': 'old 1',
+                        'old_value_integer': cls.user_admin.id,
+                    }),
+                ]
+            }
+            for msg_idx in range(2)
+            for record_idx, record in enumerate(cls.record_ratings)
+        ])
+
+        cls.messages_records = [cls.env[message.model].browse(message.res_id) for message in cls.messages_all]
```

Comment:
`browse([message.res_id for message in cls.message_all])`
I guess these are already in cache anyway so just a nitpick